### PR TITLE
feat: override jumpstart content bucket

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
feat: override jumpstart content bucket. This feature allows developers to test jumpstart specs/manifest files that are not in production, but rather hosted in an arbitrary s3 bucket. 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/utils.py [hunk 1]
```diff
@@ -60,6 +61,12 @@ def get_jumpstart_content_bucket(region: str) -> str:
     Raises:
         RuntimeError: If JumpStart is not launched in ``region``.
     """
+
+    if (
+        constants.ENV_VARIABLE_JUMPSTART_CONTENT_BUCKET_OVERRIDE in os.environ
+        and len(os.environ[constants.ENV_VARIABLE_JUMPSTART_CONTENT_BUCKET_OVERRIDE]) > 0
+    ):
+        return os.environ[constants.ENV_VARIABLE_JUMPSTART_CONTENT_BUCKET_OVERRIDE]
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/utils.py:69]: question: do you think we should log a warning if there is an override?

**Review by JGuinegagne — APPROVED** [review]
approving: just one question

File: src/sagemaker/jumpstart/utils.py (hunk 1 above)
Comment (evakravi) [reply src/sagemaker/jumpstart/utils.py:69]: It's not really a warning -- i.e. we shouldn't attach negative connotation to this. Maybe an `INFO`?

Comment (JGuinegagne) [reply src/sagemaker/jumpstart/utils.py:69]: sounds good, i agree an `INFO` log is more appropriate

**Review by JGuinegagne — APPROVED** [review]

**Review by shreyapandit — APPROVED** [review]

File: src/sagemaker/jumpstart/utils.py [hunk 2]
```diff
@@ -60,6 +61,14 @@ def get_jumpstart_content_bucket(region: str) -> str:
     Raises:
         RuntimeError: If JumpStart is not launched in ``region``.
```
Comment (shreyapandit) [inline src/sagemaker/jumpstart/utils.py:62]: Where is this used downstream and what will happen for multiple values?

**Review by navinsoni — APPROVED** [review]

**Pull request merged**